src/Microsoft.OData.Core/ODataMessageWriter.cs

```diff
@@ -842,15 +843,15 @@ private void EnsureODataFormatAndContentType()
 
                     //// TODO: What about the encoding - should we verify that it's 7bit US-ASCII only?
 
-                    this.batchBoundary = ODataBatchWriterUtils.CreateBatchBoundary(this.writingResponse);
+                    this.batchBoundary = ODataMultipartMixedBatchWriterUtils.CreateBatchBoundary(this.writingResponse);
 
                     // Set the content type header here since all headers have to be set before getting the stream
                     // Note that the mediaType may have additional parameters, which we ignore here (intentional as per MIME spec).
                     // Note that we always generate a new boundary string here, even if the accept header contained one.
                     // We need the boundary to be as unique as possible to avoid possible collision with content of the batch operation payload.
                     // Our boundary string are generated to fulfill this requirement, client specified ones might not which might lead to wrong responses
                     // and at least in theory security issues.
-                    contentType = ODataBatchWriterUtils.CreateMultipartMixedContentType(this.batchBoundary);
+                    contentType = ODataMultipartMixedBatchWriterUtils.CreateMultipartMixedContentType(this.batchBoundary);
                 }

```

Comment:
This doesn't seem like it should be in the base ODataBatchWriter class.  The base class shouldn't know anything about batch boundaries.

Reply:
see above.

---
In reply to: [145853390](https://github.com/OData/odata.net/pull/960#discussion_r145853390) [](ancestors = 145853390)

Reply:
MessageWriter should not know about format-specific details like batch boundaries. See separate thread for suggested refactoring.

---
In reply to: [146076874](https://github.com/OData/odata.net/pull/960#discussion_r146076874) [](ancestors = 146076874,145853390)